server/routes.go

```diff
@@ -125,24 +135,24 @@ func (r *routesImpl) SetDefaultRoute(backend string) {
 	}).Info("Using default route")
 }
 
-func (r *routesImpl) FindBackendForServerAddress(serverAddress string) (string, string) {
+func (r *routesImpl) FindBackendForServerAddress(ctx context.Context, serverAddress string) (string, string) {
 	r.RLock()
 	defer r.RUnlock()
 
 	addressParts := strings.Split(serverAddress, "\x00")
 
 	address := strings.ToLower(addressParts[0])
 
-	if r.mappings == nil {
-		return r.defaultRoute, address
-	} else {
-
-		if route, exists := r.mappings[address]; exists {
-			return route, address
-		} else {
-			return r.defaultRoute, address
+	if r.mappings != nil {
+		if mapping, exists := r.mappings[address]; exists {
+			if err := mapping.waker(ctx); err == nil {
```

Comment:
I would prefer that the action of waking the backend happen outside of this function since a "find" function really shouldn't have any side effects. I'd say have this function return `(string, string, func(ctx context.Context) error)` and then have `connectorImpl.findAndConnectBackend` call it.

Reply:
My comment still needs to be addressed.

Reply:
Done. (The disadvantage of doing it in `findAndConnectBackend()` instead of here in `FindBackendForServerAddress` is that we can not (easily, as-is) fall back to the default route anymore if wake up fails.)